--- ivykis-0.42.3/src/pthr.h~	2018-05-04 08:05:07.000000000 +0200
+++ ivykis-0.42.3/src/pthr.h	2019-02-09 20:05:14.835849535 +0100
@@ -42,7 +42,7 @@
  * symbol because that causes it to be undefined even if you link
  * libpthread_nonshared.a in explicitly.
  */
-#ifndef HAVE_LIBPTHREAD_NONSHARED
+#if !defined(HAVE_LIBPTHREAD_NONSHARED) && !defined(__ILP32__)
 #pragma weak pthread_atfork
 #endif
 
